### PR TITLE
Give unsafe extensions their own home.

### PR DESCRIPTION
**Pull request by mrvoorhe**
The managed tests still have gc problems.  This just organizes the problematic code a little more nicely.

### CODE REVIEW

File: unity/unity-embed-host/UnsafeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+/// </summary>
+static class UnsafeExtensions
+{
+    public static object ToManagedRepresentation(this nint intPtr)
+        => Unsafe.As<nint, object>(ref intPtr);
+
+    public static nint ToNativeRepresentation(this object obj)
+        => obj.ToNativeRepresentation<nint>();
+
+    public static T ToNativeRepresentation<T>(this object obj)
```
Comment (joncham) [inline unity/unity-embed-host/UnsafeExtensions.cs]: Nit, does any code need this public generic version?

**Review by joncham — APPROVED** [review]

File: unity/unity-embed-host/UnsafeExtensions.cs (hunk 1 above)
Comment (mrvoorhe) [reply unity/unity-embed-host/UnsafeExtensions.cs]: No.  Now that you mention it, I don't need it's needed at all.  I removed it.

Comment (UnityAlex) [reply unity/unity-embed-host/UnsafeExtensions.cs]: Also nit: "Representation" is a little wordy. Could we have `ToManaged` and `ToNative` ?

**Review by UnityAlex — APPROVED** [review]

File: unity/unity-embed-host/UnsafeExtensions.cs (hunk 1 above)
Comment (joncham) [reply unity/unity-embed-host/UnsafeExtensions.cs]: I'll add my opinion too ;-) maybe there is a better word, but I am okay with more verbosity for these specific APIs as I want to make sure whoever is calling them understands what the implications are.

Comment (mrvoorhe) [reply unity/unity-embed-host/UnsafeExtensions.cs]: I went wordy for the reason @joncham brought up.   I'm happy to change it to something else, but I do a agree that `ToManaged` and `ToNative` are too simple

**Pull request merged**